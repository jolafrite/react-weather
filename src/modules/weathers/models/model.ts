export interface IWeather {
  id: number;
  name: string;
  country: string;
  dt: number;
  dt_txt: string;
  date: string;
  time: string;
  main: IWeatherMain;
  weather: IWeatherWeather;
  wind: IWeatherWind;
}

export interface IWeatherForecast {
  id: number;
  forecast: IWeatherDailyForecast;
}

export interface IWeatherDailyForecast {
  [day: string]: IWeatherHourlyForecast;
}

export interface IWeatherHourlyForecast {
  [hour: string]: IWeatherForecastDetails;
}

export interface IWeatherForecastDetails {
  period: string;
  main: IWeatherMain;
  weather: IWeatherWeather;
  wind: IWeatherWind;
}

export interface IWeatherPredictedDaysForecast {
  [day: string]: IWeatherForecastDetails;
}

export interface IWeatherMain {
  temp: number;
  temp_min: number;
  temp_max: number;
  pressure?: number;
  humidity?: number;
  sea_level?: number;
  grnd_level?: number;
  temp_kf?: number;
}

export interface IWeatherWeather {
  id: number;
  main: string;
  description: string;
  icon: string;
}

export interface IWeatherWind {
  speed: number;
  deg: number;
  gust?: number
}

export interface IWeatherClouds {
  all: number
}

export interface IWeatherRain {
  "1h"?: number;
  "3h"?: number;
}

export interface IWeatherSnow {
  "1h"?: number;
  "3h"?: number;
};

export type IWeatherForecastType = 'day' | 'hour';
