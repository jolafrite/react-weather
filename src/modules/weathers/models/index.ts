export * from './fixtures';
export * from './model';
