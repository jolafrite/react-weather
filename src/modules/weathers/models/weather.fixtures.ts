import { useFixtureFactory } from "../../../common/utils/test";
import { IWeather } from "./model";
import { ISearchWeather } from "../../../fetch/search-weather/model";

export const weatherFixtures = useFixtureFactory<IWeather>({
  city: {
    id: 1851632,
    name: "Shuzenji",
    coord: { lon: 138.933334, lat: 34.966671 },
    country: "JP"
  },
  cnt: 38,
  list: [
    {
      dt: 1406106000,
      main: {
        temp: 298.77,
        temp_min: 298.77,
        temp_max: 298.774,
        pressure: 1005.93,
        sea_level: 1018.18,
        grnd_level: 1005.93,
        humidity: 87,
        temp_kf: 0.26
      },
      weather: [
        { id: 804, main: "Clouds", description: "overcast clouds", icon: "04d" }
      ],
      clouds: { all: 88 },
      wind: { speed: 5.71, deg: 229.501 },
      dt_txt: "2014-07-23 09:00:00"
    }
  ]
});

export const searchWeatherFixtures = useFixtureFactory<ISearchWeather>({
  id: 2643743,
  name: "London",
  coord: { lat: 51.5085, lon: -0.1258 },
  main: {
    temp: 284.31,
    pressure: 1010,
    humidity: 87,
    temp_min: 283.15,
    temp_max: 285.93,
    sea_level: 1,
    grnd_level: 2,
    temp_kf: 3,
  },
  dt: 1560233226,
  wind: {
    speed: 1.5,
    deg: 150
  },
  sys: {
    country: "GB"
  },
  rain: {
    "1h": 3,
    "3h": 4,
  },
  snow: {
    "1h": 1,
    "3h": 2,
  },
  clouds: { all: 90 },
  weather: [
    { id: 500, main: "Rain", description: "light rain", icon: "10d" },
    { id: 701, main: "Mist", description: "mist", icon: "50d" }
  ]
});
