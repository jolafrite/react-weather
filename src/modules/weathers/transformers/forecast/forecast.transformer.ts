import { ILoad5DaysWeatherForecastFetchResponse } from "../../fetch";
import {
  IWeatherDailyForecast,
  IWeatherForecast,
  IWeatherForecastDetails
} from "../../models";
import { pick } from "../../../../common/utils/object";

const hourForecastTransformer = (data: any): IWeatherForecastDetails => {
  const [date, time] = data.dt_txt.split(" ");

  return {
    ...pick(data, ["main", "weather", "wind"]),
    weather: data.weather && data.weather[0],
    period: time,
  };
};

export const load5DaysWeatherForecastTransformer = (
  data: ILoad5DaysWeatherForecastFetchResponse
): IWeatherForecast => {
  const response = data.list.reduce(
    (saved, item) => {
      const [date, hour] = item.dt_txt.split(" ");

      return {
        ...saved,
        [date]: {
          ...saved[date],
          [hour]: hourForecastTransformer(item)
        }
      };
    },
    {} as IWeatherDailyForecast
  );

  return {
    id: data.city.id,
    forecast: response
  };
};
