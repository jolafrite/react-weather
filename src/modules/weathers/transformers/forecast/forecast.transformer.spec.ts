import { load5DaysWeatherForecastFetchResponseFixture } from "../../models";
import { load5DaysWeatherForecastTransformer } from "./forecast.transformer";

describe("transformer: Forecast", () => {
  describe("load5DaysWeatherForecastTransformer", () => {
    test("should convert the response into a list of IWeather", () => {
      const fixture = load5DaysWeatherForecastFetchResponseFixture();

      expect(load5DaysWeatherForecastTransformer(fixture)).toMatchSnapshot();
    });
  });

});
