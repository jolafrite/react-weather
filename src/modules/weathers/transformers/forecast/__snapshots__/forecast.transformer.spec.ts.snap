// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformer: Forecast load5DaysWeatherForecastTransformer should convert the response into a list of IWeather 1`] = `
Object {
  "2019-06-16": Object {
    "09:00:00": Object {
      "date": "2019-06-16",
      "dt": 1560675600,
      "dt_txt": "2019-06-16 09:00:00",
      "main": Object {
        "grnd_level": 998.17,
        "humidity": 63,
        "pressure": 1004.58,
        "sea_level": 1004.58,
        "temp": 28.61,
        "temp_kf": -0.46,
        "temp_max": 29.07,
        "temp_min": 28.61,
      },
      "time": "09:00:00",
      "weather": Object {
        "description": "broken clouds",
        "icon": "04d",
        "id": 803,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 111.264,
        "speed": 5.7,
      },
    },
    "12:00:00": Object {
      "date": "2019-06-16",
      "dt": 1560686400,
      "dt_txt": "2019-06-16 12:00:00",
      "main": Object {
        "grnd_level": 999.93,
        "humidity": 79,
        "pressure": 1006.33,
        "sea_level": 1006.33,
        "temp": 26.44,
        "temp_kf": -0.35,
        "temp_max": 26.79,
        "temp_min": 26.44,
      },
      "time": "12:00:00",
      "weather": Object {
        "description": "broken clouds",
        "icon": "04n",
        "id": 803,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 110.461,
        "speed": 5.64,
      },
    },
    "15:00:00": Object {
      "date": "2019-06-16",
      "dt": 1560697200,
      "dt_txt": "2019-06-16 15:00:00",
      "main": Object {
        "grnd_level": 1001.02,
        "humidity": 79,
        "pressure": 1007.65,
        "sea_level": 1007.65,
        "temp": 26.68,
        "temp_kf": -0.23,
        "temp_max": 26.91,
        "temp_min": 26.68,
      },
      "time": "15:00:00",
      "weather": Object {
        "description": "broken clouds",
        "icon": "04n",
        "id": 803,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 105.122,
        "speed": 6.15,
      },
    },
    "18:00:00": Object {
      "date": "2019-06-16",
      "dt": 1560708000,
      "dt_txt": "2019-06-16 18:00:00",
      "main": Object {
        "grnd_level": 1000.05,
        "humidity": 81,
        "pressure": 1006.43,
        "sea_level": 1006.43,
        "temp": 26.8,
        "temp_kf": -0.12,
        "temp_max": 26.92,
        "temp_min": 26.8,
      },
      "time": "18:00:00",
      "weather": Object {
        "description": "broken clouds",
        "icon": "04n",
        "id": 803,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 96.253,
        "speed": 4.31,
      },
    },
    "21:00:00": Object {
      "date": "2019-06-16",
      "dt": 1560718800,
      "dt_txt": "2019-06-16 21:00:00",
      "main": Object {
        "grnd_level": 999.33,
        "humidity": 81,
        "pressure": 1005.99,
        "sea_level": 1005.99,
        "temp": 26.43,
        "temp_kf": 0,
        "temp_max": 26.43,
        "temp_min": 26.43,
      },
      "time": "21:00:00",
      "weather": Object {
        "description": "overcast clouds",
        "icon": "04n",
        "id": 804,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 91.63,
        "speed": 4.19,
      },
    },
  },
  "2019-06-17": Object {
    "00:00:00": Object {
      "date": "2019-06-17",
      "dt": 1560729600,
      "dt_txt": "2019-06-17 00:00:00",
      "main": Object {
        "grnd_level": 1000.73,
        "humidity": 78,
        "pressure": 1007.16,
        "sea_level": 1007.16,
        "temp": 27.15,
        "temp_kf": 0,
        "temp_max": 27.15,
        "temp_min": 27.15,
      },
      "time": "00:00:00",
      "weather": Object {
        "description": "overcast clouds",
        "icon": "04d",
        "id": 804,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 82.081,
        "speed": 4.38,
      },
    },
    "03:00:00": Object {
      "date": "2019-06-17",
      "dt": 1560740400,
      "dt_txt": "2019-06-17 03:00:00",
      "main": Object {
        "grnd_level": 1001.08,
        "humidity": 70,
        "pressure": 1007.49,
        "sea_level": 1007.49,
        "temp": 29.09,
        "temp_kf": 0,
        "temp_max": 29.09,
        "temp_min": 29.09,
      },
      "time": "03:00:00",
      "weather": Object {
        "description": "overcast clouds",
        "icon": "04d",
        "id": 804,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 89.503,
        "speed": 4.61,
      },
    },
    "06:00:00": Object {
      "date": "2019-06-17",
      "dt": 1560751200,
      "dt_txt": "2019-06-17 06:00:00",
      "main": Object {
        "grnd_level": 999.59,
        "humidity": 67,
        "pressure": 1006.26,
        "sea_level": 1006.26,
        "temp": 30.8,
        "temp_kf": 0,
        "temp_max": 30.8,
        "temp_min": 30.8,
      },
      "time": "06:00:00",
      "weather": Object {
        "description": "overcast clouds",
        "icon": "04d",
        "id": 804,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 103.161,
        "speed": 5.27,
      },
    },
    "09:00:00": Object {
      "date": "2019-06-17",
      "dt": 1560762000,
      "dt_txt": "2019-06-17 09:00:00",
      "main": Object {
        "grnd_level": 998.64,
        "humidity": 73,
        "pressure": 1005.27,
        "sea_level": 1005.27,
        "temp": 29.53,
        "temp_kf": 0,
        "temp_max": 29.53,
        "temp_min": 29.53,
      },
      "time": "09:00:00",
      "weather": Object {
        "description": "moderate rain",
        "icon": "10d",
        "id": 501,
        "main": "Rain",
      },
      "wind": Object {
        "deg": 109.551,
        "speed": 6.3,
      },
    },
    "12:00:00": Object {
      "date": "2019-06-17",
      "dt": 1560772800,
      "dt_txt": "2019-06-17 12:00:00",
      "main": Object {
        "grnd_level": 1000.69,
        "humidity": 81,
        "pressure": 1007.11,
        "sea_level": 1007.11,
        "temp": 27.85,
        "temp_kf": 0,
        "temp_max": 27.85,
        "temp_min": 27.85,
      },
      "time": "12:00:00",
      "weather": Object {
        "description": "light rain",
        "icon": "10n",
        "id": 500,
        "main": "Rain",
      },
      "wind": Object {
        "deg": 107.714,
        "speed": 5.5,
      },
    },
    "15:00:00": Object {
      "date": "2019-06-17",
      "dt": 1560783600,
      "dt_txt": "2019-06-17 15:00:00",
      "main": Object {
        "grnd_level": 1002.09,
        "humidity": 84,
        "pressure": 1008.64,
        "sea_level": 1008.64,
        "temp": 27.43,
        "temp_kf": 0,
        "temp_max": 27.43,
        "temp_min": 27.43,
      },
      "time": "15:00:00",
      "weather": Object {
        "description": "overcast clouds",
        "icon": "04n",
        "id": 804,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 108.559,
        "speed": 4.1,
      },
    },
    "18:00:00": Object {
      "date": "2019-06-17",
      "dt": 1560794400,
      "dt_txt": "2019-06-17 18:00:00",
      "main": Object {
        "grnd_level": 1000.89,
        "humidity": 83,
        "pressure": 1007.42,
        "sea_level": 1007.42,
        "temp": 27.4,
        "temp_kf": 0,
        "temp_max": 27.4,
        "temp_min": 27.4,
      },
      "time": "18:00:00",
      "weather": Object {
        "description": "light rain",
        "icon": "10n",
        "id": 500,
        "main": "Rain",
      },
      "wind": Object {
        "deg": 102.742,
        "speed": 3.62,
      },
    },
    "21:00:00": Object {
      "date": "2019-06-17",
      "dt": 1560805200,
      "dt_txt": "2019-06-17 21:00:00",
      "main": Object {
        "grnd_level": 1000.49,
        "humidity": 85,
        "pressure": 1007.08,
        "sea_level": 1007.08,
        "temp": 26.94,
        "temp_kf": 0,
        "temp_max": 26.94,
        "temp_min": 26.94,
      },
      "time": "21:00:00",
      "weather": Object {
        "description": "light rain",
        "icon": "10n",
        "id": 500,
        "main": "Rain",
      },
      "wind": Object {
        "deg": 101.444,
        "speed": 4.52,
      },
    },
  },
  "2019-06-18": Object {
    "00:00:00": Object {
      "date": "2019-06-18",
      "dt": 1560816000,
      "dt_txt": "2019-06-18 00:00:00",
      "main": Object {
        "grnd_level": 1001.29,
        "humidity": 78,
        "pressure": 1007.85,
        "sea_level": 1007.85,
        "temp": 28.15,
        "temp_kf": 0,
        "temp_max": 28.15,
        "temp_min": 28.15,
      },
      "time": "00:00:00",
      "weather": Object {
        "description": "light rain",
        "icon": "10d",
        "id": 500,
        "main": "Rain",
      },
      "wind": Object {
        "deg": 96.245,
        "speed": 4.44,
      },
    },
    "03:00:00": Object {
      "date": "2019-06-18",
      "dt": 1560826800,
      "dt_txt": "2019-06-18 03:00:00",
      "main": Object {
        "grnd_level": 1001.7,
        "humidity": 68,
        "pressure": 1008.06,
        "sea_level": 1008.06,
        "temp": 30.68,
        "temp_kf": 0,
        "temp_max": 30.68,
        "temp_min": 30.68,
      },
      "time": "03:00:00",
      "weather": Object {
        "description": "light rain",
        "icon": "10d",
        "id": 500,
        "main": "Rain",
      },
      "wind": Object {
        "deg": 122.994,
        "speed": 4.45,
      },
    },
    "06:00:00": Object {
      "date": "2019-06-18",
      "dt": 1560837600,
      "dt_txt": "2019-06-18 06:00:00",
      "main": Object {
        "grnd_level": 999.71,
        "humidity": 62,
        "pressure": 1005.95,
        "sea_level": 1005.95,
        "temp": 31.68,
        "temp_kf": 0,
        "temp_max": 31.68,
        "temp_min": 31.68,
      },
      "time": "06:00:00",
      "weather": Object {
        "description": "light rain",
        "icon": "10d",
        "id": 500,
        "main": "Rain",
      },
      "wind": Object {
        "deg": 161.414,
        "speed": 5.99,
      },
    },
    "09:00:00": Object {
      "date": "2019-06-18",
      "dt": 1560848400,
      "dt_txt": "2019-06-18 09:00:00",
      "main": Object {
        "grnd_level": 999.11,
        "humidity": 69,
        "pressure": 1005.19,
        "sea_level": 1005.19,
        "temp": 30.23,
        "temp_kf": 0,
        "temp_max": 30.23,
        "temp_min": 30.23,
      },
      "time": "09:00:00",
      "weather": Object {
        "description": "overcast clouds",
        "icon": "04d",
        "id": 804,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 165.596,
        "speed": 5.33,
      },
    },
    "12:00:00": Object {
      "date": "2019-06-18",
      "dt": 1560859200,
      "dt_txt": "2019-06-18 12:00:00",
      "main": Object {
        "grnd_level": 1000.2,
        "humidity": 75,
        "pressure": 1006.35,
        "sea_level": 1006.35,
        "temp": 29.18,
        "temp_kf": 0,
        "temp_max": 29.18,
        "temp_min": 29.18,
      },
      "time": "12:00:00",
      "weather": Object {
        "description": "overcast clouds",
        "icon": "04n",
        "id": 804,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 146.015,
        "speed": 4.04,
      },
    },
    "15:00:00": Object {
      "date": "2019-06-18",
      "dt": 1560870000,
      "dt_txt": "2019-06-18 15:00:00",
      "main": Object {
        "grnd_level": 1001.24,
        "humidity": 79,
        "pressure": 1007.56,
        "sea_level": 1007.56,
        "temp": 28.69,
        "temp_kf": 0,
        "temp_max": 28.69,
        "temp_min": 28.69,
      },
      "time": "15:00:00",
      "weather": Object {
        "description": "overcast clouds",
        "icon": "04n",
        "id": 804,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 145.724,
        "speed": 4.07,
      },
    },
    "18:00:00": Object {
      "date": "2019-06-18",
      "dt": 1560880800,
      "dt_txt": "2019-06-18 18:00:00",
      "main": Object {
        "grnd_level": 1000.29,
        "humidity": 81,
        "pressure": 1006.55,
        "sea_level": 1006.55,
        "temp": 28.35,
        "temp_kf": 0,
        "temp_max": 28.35,
        "temp_min": 28.35,
      },
      "time": "18:00:00",
      "weather": Object {
        "description": "overcast clouds",
        "icon": "04n",
        "id": 804,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 145.245,
        "speed": 3.43,
      },
    },
    "21:00:00": Object {
      "date": "2019-06-18",
      "dt": 1560891600,
      "dt_txt": "2019-06-18 21:00:00",
      "main": Object {
        "grnd_level": 1000.27,
        "humidity": 83,
        "pressure": 1006.52,
        "sea_level": 1006.52,
        "temp": 28.17,
        "temp_kf": 0,
        "temp_max": 28.17,
        "temp_min": 28.17,
      },
      "time": "21:00:00",
      "weather": Object {
        "description": "light rain",
        "icon": "10n",
        "id": 500,
        "main": "Rain",
      },
      "wind": Object {
        "deg": 142.077,
        "speed": 2.63,
      },
    },
  },
  "2019-06-19": Object {
    "00:00:00": Object {
      "date": "2019-06-19",
      "dt": 1560902400,
      "dt_txt": "2019-06-19 00:00:00",
      "main": Object {
        "grnd_level": 1001.08,
        "humidity": 74,
        "pressure": 1007.38,
        "sea_level": 1007.38,
        "temp": 29.23,
        "temp_kf": 0,
        "temp_max": 29.23,
        "temp_min": 29.23,
      },
      "time": "00:00:00",
      "weather": Object {
        "description": "light rain",
        "icon": "10d",
        "id": 500,
        "main": "Rain",
      },
      "wind": Object {
        "deg": 156.387,
        "speed": 2.88,
      },
    },
    "03:00:00": Object {
      "date": "2019-06-19",
      "dt": 1560913200,
      "dt_txt": "2019-06-19 03:00:00",
      "main": Object {
        "grnd_level": 1001.2,
        "humidity": 66,
        "pressure": 1007.36,
        "sea_level": 1007.36,
        "temp": 31,
        "temp_kf": 0,
        "temp_max": 31,
        "temp_min": 31,
      },
      "time": "03:00:00",
      "weather": Object {
        "description": "overcast clouds",
        "icon": "04d",
        "id": 804,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 177.174,
        "speed": 4.32,
      },
    },
    "06:00:00": Object {
      "date": "2019-06-19",
      "dt": 1560924000,
      "dt_txt": "2019-06-19 06:00:00",
      "main": Object {
        "grnd_level": 999.17,
        "humidity": 67,
        "pressure": 1005.15,
        "sea_level": 1005.15,
        "temp": 31.38,
        "temp_kf": 0,
        "temp_max": 31.38,
        "temp_min": 31.38,
      },
      "time": "06:00:00",
      "weather": Object {
        "description": "overcast clouds",
        "icon": "04d",
        "id": 804,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 184.332,
        "speed": 4.83,
      },
    },
    "09:00:00": Object {
      "date": "2019-06-19",
      "dt": 1560934800,
      "dt_txt": "2019-06-19 09:00:00",
      "main": Object {
        "grnd_level": 997.61,
        "humidity": 73,
        "pressure": 1003.64,
        "sea_level": 1003.64,
        "temp": 30.38,
        "temp_kf": 0,
        "temp_max": 30.38,
        "temp_min": 30.38,
      },
      "time": "09:00:00",
      "weather": Object {
        "description": "overcast clouds",
        "icon": "04d",
        "id": 804,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 176.247,
        "speed": 5.3,
      },
    },
    "12:00:00": Object {
      "date": "2019-06-19",
      "dt": 1560945600,
      "dt_txt": "2019-06-19 12:00:00",
      "main": Object {
        "grnd_level": 998.53,
        "humidity": 80,
        "pressure": 1004.48,
        "sea_level": 1004.48,
        "temp": 29.05,
        "temp_kf": 0,
        "temp_max": 29.05,
        "temp_min": 29.05,
      },
      "time": "12:00:00",
      "weather": Object {
        "description": "overcast clouds",
        "icon": "04n",
        "id": 804,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 169.942,
        "speed": 4.19,
      },
    },
    "15:00:00": Object {
      "date": "2019-06-19",
      "dt": 1560956400,
      "dt_txt": "2019-06-19 15:00:00",
      "main": Object {
        "grnd_level": 1000.09,
        "humidity": 84,
        "pressure": 1006.29,
        "sea_level": 1006.29,
        "temp": 28.65,
        "temp_kf": 0,
        "temp_max": 28.65,
        "temp_min": 28.65,
      },
      "time": "15:00:00",
      "weather": Object {
        "description": "overcast clouds",
        "icon": "04n",
        "id": 804,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 160.281,
        "speed": 4.35,
      },
    },
    "18:00:00": Object {
      "date": "2019-06-19",
      "dt": 1560967200,
      "dt_txt": "2019-06-19 18:00:00",
      "main": Object {
        "grnd_level": 998.85,
        "humidity": 84,
        "pressure": 1005.13,
        "sea_level": 1005.13,
        "temp": 28.25,
        "temp_kf": 0,
        "temp_max": 28.25,
        "temp_min": 28.25,
      },
      "time": "18:00:00",
      "weather": Object {
        "description": "overcast clouds",
        "icon": "04n",
        "id": 804,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 164.074,
        "speed": 3.66,
      },
    },
    "21:00:00": Object {
      "date": "2019-06-19",
      "dt": 1560978000,
      "dt_txt": "2019-06-19 21:00:00",
      "main": Object {
        "grnd_level": 998.77,
        "humidity": 85,
        "pressure": 1005.1,
        "sea_level": 1005.1,
        "temp": 28.15,
        "temp_kf": 0,
        "temp_max": 28.15,
        "temp_min": 28.15,
      },
      "time": "21:00:00",
      "weather": Object {
        "description": "light rain",
        "icon": "10n",
        "id": 500,
        "main": "Rain",
      },
      "wind": Object {
        "deg": 166.47,
        "speed": 3.02,
      },
    },
  },
  "2019-06-20": Object {
    "00:00:00": Object {
      "date": "2019-06-20",
      "dt": 1560988800,
      "dt_txt": "2019-06-20 00:00:00",
      "main": Object {
        "grnd_level": 999.94,
        "humidity": 79,
        "pressure": 1006.12,
        "sea_level": 1006.12,
        "temp": 29.02,
        "temp_kf": 0,
        "temp_max": 29.02,
        "temp_min": 29.02,
      },
      "time": "00:00:00",
      "weather": Object {
        "description": "light rain",
        "icon": "10d",
        "id": 500,
        "main": "Rain",
      },
      "wind": Object {
        "deg": 190.618,
        "speed": 4.37,
      },
    },
    "03:00:00": Object {
      "date": "2019-06-20",
      "dt": 1560999600,
      "dt_txt": "2019-06-20 03:00:00",
      "main": Object {
        "grnd_level": 1000.26,
        "humidity": 72,
        "pressure": 1006.32,
        "sea_level": 1006.32,
        "temp": 30.35,
        "temp_kf": 0,
        "temp_max": 30.35,
        "temp_min": 30.35,
      },
      "time": "03:00:00",
      "weather": Object {
        "description": "overcast clouds",
        "icon": "04d",
        "id": 804,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 196.868,
        "speed": 5.17,
      },
    },
    "06:00:00": Object {
      "date": "2019-06-20",
      "dt": 1561010400,
      "dt_txt": "2019-06-20 06:00:00",
      "main": Object {
        "grnd_level": 997.98,
        "humidity": 68,
        "pressure": 1004.05,
        "sea_level": 1004.05,
        "temp": 31.25,
        "temp_kf": 0,
        "temp_max": 31.25,
        "temp_min": 31.25,
      },
      "time": "06:00:00",
      "weather": Object {
        "description": "overcast clouds",
        "icon": "04d",
        "id": 804,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 192.522,
        "speed": 5.14,
      },
    },
    "09:00:00": Object {
      "date": "2019-06-20",
      "dt": 1561021200,
      "dt_txt": "2019-06-20 09:00:00",
      "main": Object {
        "grnd_level": 996.63,
        "humidity": 73,
        "pressure": 1002.68,
        "sea_level": 1002.68,
        "temp": 30.65,
        "temp_kf": 0,
        "temp_max": 30.65,
        "temp_min": 30.65,
      },
      "time": "09:00:00",
      "weather": Object {
        "description": "broken clouds",
        "icon": "04d",
        "id": 803,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 186.912,
        "speed": 5.03,
      },
    },
    "12:00:00": Object {
      "date": "2019-06-20",
      "dt": 1561032000,
      "dt_txt": "2019-06-20 12:00:00",
      "main": Object {
        "grnd_level": 997.63,
        "humidity": 79,
        "pressure": 1003.8,
        "sea_level": 1003.8,
        "temp": 29.43,
        "temp_kf": 0,
        "temp_max": 29.43,
        "temp_min": 29.43,
      },
      "time": "12:00:00",
      "weather": Object {
        "description": "broken clouds",
        "icon": "04n",
        "id": 803,
        "main": "Clouds",
      },
      "wind": Object {
        "deg": 175.339,
        "speed": 4.43,
      },
    },
    "15:00:00": Object {
      "date": "2019-06-20",
      "dt": 1561042800,
      "dt_txt": "2019-06-20 15:00:00",
      "main": Object {
        "grnd_level": 998.67,
        "humidity": 84,
        "pressure": 1004.87,
        "sea_level": 1004.87,
        "temp": 28.87,
        "temp_kf": 0,
        "temp_max": 28.87,
        "temp_min": 28.87,
      },
      "time": "15:00:00",
      "weather": Object {
        "description": "light rain",
        "icon": "10n",
        "id": 500,
        "main": "Rain",
      },
      "wind": Object {
        "deg": 178.277,
        "speed": 4.56,
      },
    },
    "18:00:00": Object {
      "date": "2019-06-20",
      "dt": 1561053600,
      "dt_txt": "2019-06-20 18:00:00",
      "main": Object {
        "grnd_level": 997.52,
        "humidity": 85,
        "pressure": 1003.8,
        "sea_level": 1003.8,
        "temp": 28.45,
        "temp_kf": 0,
        "temp_max": 28.45,
        "temp_min": 28.45,
      },
      "time": "18:00:00",
      "weather": Object {
        "description": "light rain",
        "icon": "10n",
        "id": 500,
        "main": "Rain",
      },
      "wind": Object {
        "deg": 189.193,
        "speed": 4.94,
      },
    },
    "21:00:00": Object {
      "date": "2019-06-20",
      "dt": 1561064400,
      "dt_txt": "2019-06-20 21:00:00",
      "main": Object {
        "grnd_level": 997.51,
        "humidity": 83,
        "pressure": 1003.73,
        "sea_level": 1003.73,
        "temp": 28.28,
        "temp_kf": 0,
        "temp_max": 28.28,
        "temp_min": 28.28,
      },
      "time": "21:00:00",
      "weather": Object {
        "description": "light rain",
        "icon": "10n",
        "id": 500,
        "main": "Rain",
      },
      "wind": Object {
        "deg": 203.588,
        "speed": 5.75,
      },
    },
  },
  "2019-06-21": Object {
    "00:00:00": Object {
      "date": "2019-06-21",
      "dt": 1561075200,
      "dt_txt": "2019-06-21 00:00:00",
      "main": Object {
        "grnd_level": 998.52,
        "humidity": 80,
        "pressure": 1004.77,
        "sea_level": 1004.77,
        "temp": 29.08,
        "temp_kf": 0,
        "temp_max": 29.08,
        "temp_min": 29.08,
      },
      "time": "00:00:00",
      "weather": Object {
        "description": "light rain",
        "icon": "10d",
        "id": 500,
        "main": "Rain",
      },
      "wind": Object {
        "deg": 199.832,
        "speed": 4.9,
      },
    },
    "03:00:00": Object {
      "date": "2019-06-21",
      "dt": 1561086000,
      "dt_txt": "2019-06-21 03:00:00",
      "main": Object {
        "grnd_level": 999.26,
        "humidity": 75,
        "pressure": 1005.17,
        "sea_level": 1005.17,
        "temp": 30.35,
        "temp_kf": 0,
        "temp_max": 30.35,
        "temp_min": 30.35,
      },
      "time": "03:00:00",
      "weather": Object {
        "description": "light rain",
        "icon": "10d",
        "id": 500,
        "main": "Rain",
      },
      "wind": Object {
        "deg": 204.114,
        "speed": 6.12,
      },
    },
    "06:00:00": Object {
      "date": "2019-06-21",
      "dt": 1561096800,
      "dt_txt": "2019-06-21 06:00:00",
      "main": Object {
        "grnd_level": 997.31,
        "humidity": 74,
        "pressure": 1003.05,
        "sea_level": 1003.05,
        "temp": 30.57,
        "temp_kf": 0,
        "temp_max": 30.57,
        "temp_min": 30.57,
      },
      "time": "06:00:00",
      "weather": Object {
        "description": "light rain",
        "icon": "10d",
        "id": 500,
        "main": "Rain",
      },
      "wind": Object {
        "deg": 198.667,
        "speed": 6.24,
      },
    },
  },
}
`;
